### ISSUE TITLE
Tier 1 Support for x86_64 macOS

### ISSUE THREAD

**Issue by andrewrk** [ziglang/zig#4897]
I just demoted macOS from Tier 1 to Tier 2 support on the [support table](https://ziglang.org/#Support-Table), because the requirements for Tier 1 say this:

> libc is available for this target even when cross compiling.

Here are some of the roadblocks for cross-compiling targeting macOS:

 * ~LLD linker macho code is unmaintained~ [this is actually resolved now](http://lists.llvm.org/pipermail/llvm-dev/2020-March/139641.html)
 * darwin frameworks (#1349)
 * Apple has not been careful with the license in their header files and it does not look like we can ship their .h files. So we would need a clean room implementation of darwin libc header files.
 * The requirement for applications to be signed causes problems

**Comment by robinsoncol** [comment]
I couldn't get the latest version of Zig to build on macOS Catalina. Is the build problem that I'm having related to this issue?

Here's the error that I'm getting:
```
[ 99%] Linking CXX executable zig0
Undefined symbols for architecture x86_64:
  "getPollyPluginInfo()", referenced from:
      (anonymous namespace)::EmitAssemblyHelper::EmitAssemblyWithNewPassManager(clang::BackendAction, std::__1::unique_ptr<llvm::raw_pwrite_stream, std::__1::default_delete<llvm::raw_pwrite_stream> >) in libclangCodeGen.a(BackendUtil.cpp.o)
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [zig0] Error 1
make[1]: *** [CMakeFiles/zig0.dir/all] Error 2
make: *** [all] Error 2
```

cmake 3.17.0_1
llvm 10.0

**Comment by fengb** [comment]
@robinsoncol Homebrew’s LLVM is misconfigured: https://github.com/Homebrew/homebrew-core/pull/52352

**Comment by kubkon** [comment]
@andrewrk can we revisit this? Out of the list posted in the issue's description:

> * ~~libc is available for this target even when cross compiling~~
> * ~~LLD linker macho code is unmaintained this is actually resolved now~~
> * **darwin frameworks (#1349)**
> * ~~Apple has not been careful with the license in their header files and it does not look like we can ship their .h files. So we would need a clean room implementation of darwin libc header files.~~
> * ~~The requirement for applications to be signed causes problems~~

only the 3 bullet is still relevant (and probably will be for quite some time); however, the rest seems addressed by now including shipping libc headers for both `aarch64-macos-gnu` and `x86_64-macos-gnu` targets. In the light of this plus the fact that the work on self-hosted Mach-O linker is progressing at a steady pace, I'd like to propose we promote macOS back to Tier 1 for `x86_64` and introduce `arm64` as Tier 2. If there are no objections, I'll be happy to submit relevant changes to the website.

**Comment by andrewrk** [comment]
I agree, thanks to your diligent work, zig now has tier 1 support for x86_64-macos! Would you like to do the honors of closing this issue?

**Comment by kubkon** [comment]
> I agree, thanks to your diligent work, zig now has tier 1 support for x86_64-macos! Would you like to do the honors of closing this issue?

Oh absolutely, thanks!

**Comment by kubkon** [comment]
PR upgrading `x86_64` to tier 1, and introducing `arm64` as tier 2: https://github.com/ziglang/www.ziglang.org/pull/86

**Comment by alexrp** [comment]
We now consider self-hosted backend support to be a criteria for Tier 1.
